File: pdu/python/drone_srv_msgs/pdu_cdr_conv_LiDARScanRequest.py
from .pdu_pytype_LiDARScanRequest import LiDARScanRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_LiDARScanRequest(writer: CdrWriter, src: LiDARScanRequest):
    writer.write_string(src.drone_name)


def cdr_body_to_py_LiDARScanRequest(reader: CdrReader, dst: LiDARScanRequest):
    dst.drone_name = reader.read_string()
    return dst


def py_to_cdr_LiDARScanRequest(src: LiDARScanRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_LiDARScanRequest(writer, src)
    return writer.bytes()


def cdr_to_py_LiDARScanRequest(cdr_payload) -> LiDARScanRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = LiDARScanRequest()
    return cdr_body_to_py_LiDARScanRequest(reader, dst)

File: pdu/javascript/drone_srv_msgs/pdu_cdr_conv_LiDARScanRequest.js
import { LiDARScanRequest } from './pdu_jstype_LiDARScanRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduLiDARScanRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { LiDARScanRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_string(src.drone_name);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { LiDARScanRequest } dst
     * @returns { LiDARScanRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.drone_name = reader.read_string();
        return dst;
    }

    /**
     * @param { LiDARScanRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { LiDARScanRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new LiDARScanRequest());
    }
}
